### ISSUE TITLE
Extend source-map-based symbolizer used in msan/asan to all backtraces under node js

### ISSUE THREAD

**Issue by sbc100** [emscripten-core/emscripten#16552]
Currently under node its not possible to get anything other than wasm function names in backtraces.

With asan/msan we support parsing the source map to also get line numbers.. perhaps we could extend this to allow folks to get more detailed backtraces from node in general.

**Comment by sbc100** [comment]
@dschuff .. actually could we use your emsymbolizer work here?  Can we pipe the node output into this tool maybe?

**Comment by dschuff** [comment]
The idea is to process a stack trace (which would have an address embedded in it for each frame) and then print something with e.g. line numbers? Yeah I think think it would be straightforward to make emsymbolizer do that. That would only work for an "offline" stack trace, where e.g. as you suggested we can pipe the stack trace into emsymbolizer (or is there some way that code running in node can invoke emsymbolizer?)

The asan code is actually embedded in the emscripten JS glue, and can work on the client with no need for emcc. That's not too bad for source maps but would be harder to replicate for DWARF.

**Comment by sbc100** [comment]
I guess having the builtin approach is nice because it means you don't need to know about emsymbolizer.. on the other hand doing as an external process keeps things simple (and avoids us having to be the once that handle / print the exceptions).  Right now, we can let exceptions flow all the out of node and I'd kind of like to keep it that way.

**Comment by VirtualTim** [comment]
> Currently under node its not possible to get anything other than wasm function names in backtraces.

I'm getting the demangled c++ names when compiling with `--profiling`.
Eg (forcing a crash in NodeJS):
```
worker.js onmessage() captured an uncaught exception: RuntimeError: null function or function signature mismatch
worker sent an error! undefined:undefined: null function or function signature mismatch
Error [RuntimeError]: null function or function signature mismatch
    at NCS::ECW::CFile::CCachedBlock::UnPack(NCS::ECW::CFile::CSubBand&) const (wasm://wasm/0ad15276:wasm-function[13743]:0x3b63fb)
    at invoke_iii (eval at importScripts (/src/Source/ECWJP2Build/ECWJP2SDK_Emscripten/NCSWASMPerformanceTester/ecwsdk/Profiling/ecw61.worker.js:38:16), <anonymous>:8092:34)
    at NCS::ECW::CFile::CPrecinct::EnsureSubBand(std::__2::shared_ptr<NCS::ECW::CFile::CPrecinctViewContext> const&, NCS::ECW::CFile::CSubBand&) const (wasm://wasm/0ad15276:wasm-function[15260]:0x3daca5)
...
```

**Comment by sbc100** [comment]
Indeed, we can get function names, but the desire here is to have line numbers too.
